lib/MemoryModel/MemModel.cpp

```diff
@@ -189,7 +189,7 @@ bool SymbolTableInfo::computeGepOffset(const llvm::User *V, LocationSet& ls) {
         // Handling array types, skipe array handling here
         // We treat whole array as one, then we can distinguish different field of an array of struct
         // e.g. s[1].f1 is differet from s[0].f2
-        if(isa<ArrayType>(*gi))
+        if(isa<ArrayType>(gi.getIndexedType()))
```

Comment:
@jcarlson23 just to double-check, I think you need to 'source' the setup.sh script (`. ./setup.sh`) in order for the exported variables to persist in your current shell.

That said I'm not actually using `setup.sh` and instead just set a variety of variables directly.  Here's my Nix expression for running PTABen on a particular SVF: http://dtz.so/desac.nix

(If you're interested I can share the Nix expressions I use for SVF as well... but even if you don't know Nix it's pretty readable what's going on there I think :))

If the bridged GEP iterator /could/ be used and trusted to be compatible with previous behavior, I think that's the best solution until someone explicitly sits down and reworks some of the memory model code to use the new-style interface (which may require more extensive changes to the way it models the code, although hopefully not).

Unfortunately with something like LLVM there are often too many changes to make "diff"-based comparisons on large applications practical.  Although maybe we could get some mileage out of (ab)using the IR upgrading:

* Build large applications as LLVM 3.8 bitcode
* Run SVF 3.8 on it, dumping results to file (or something)
* Run SVF 4.0 on /exactly/ the same bitcode file and dumping analysis results similarly

That might be effective for minimizing the countless changes that would change the analysis results due to changes in clang, optimizations, etc..

Personally I'm a fan of checking analysis results dynamically--while certainly insufficient for proving correctness I've yet to find an LLVM-based pointer analysis (beyond the traditionally simple ones in LLVM) that was able to drive callgraph-checking instrumentation that didn't fail at runtime :(.